ArduPlane/tiltrotor.cpp

```diff
@@ -315,19 +330,78 @@ void Tiltrotor::continuous_update(void)
         return;
     }
 
+    float new_tilt = 0.0f;
     if (quadplane.assisted_flight &&
+        transition->transition_state == Tiltrotor_Transition::TRANSITION_AIRSPEED_WAIT &&
+        (type == TILT_TYPE_VECTORED_YAW || type == TILT_TYPE_CONTINUOUS)) {
+            const float tilt_angle_range = 90.0f * get_forward_flight_tilt() - (float)max_angle_deg;
+            if (is_positive(tilt_angle_range)) {
+                // Tilt immediately to Q_TILT_MAX. Then tilt forward from Q_TILT_MAX to the forward flight value.
+                // If when tilting forward past Q_TILT_MAX, the throttle for any tilting motor exceeds 99%, the
+                // tilt angle is reduced to prevent loss of attitude control.
+                float max_throttle = 0.0f;
+                for (uint8_t i=0; i<4; i++) {
+                    float throttle_demand;
+                    if (is_motor_tilting(i) && motors->get_motor_demand(i, throttle_demand)) {
+                        if (throttle_demand > max_throttle) {
+                            max_throttle = throttle_demand;
```

Comment:
you could stash this value in the `tilt_compensate_angle` method, that would remove the need for any helper in motors.